psl/parser-database/src/walkers.rs

```diff
@@ -45,6 +45,7 @@ impl<'db, I> PartialEq for Walker<'db, I>
 where
     I: PartialEq,
 {
+    #[allow(unconditional_recursion)]
     fn eq(&self, other: &Self) -> bool {
         self.id.eq(&other.id)
     }
```

Comment:
Clippy believes that this implementation of the `PartialEq` trait for `Walker<'_, I>` is recursive.
This is not true, as we're comparing the `id: I` fields, which are strictly contained (⊂) in `Walker<'_, I>`.

Clippy is probably unable to prove the soundness of this block since:
- `I` has no bounds other than `PartialEq`: this generic `I` could be another `Walker<'_, I>` instance, for all that clippy knows
- `db` isn't part of the comparison (by design, since a unique `crate::ParserDatabase` is used and referenced for every `Walker<'_, I>` at any given time)

It is thus safe to silence clippy's warnings via `#[allow(unconditional_recursion)]`.